addons/binding/org.openhab.binding.vitotronic/src/main/java/org/openhab/binding/vitotronic/handler/VitotronicThingHandler.java

```diff
@@ -0,0 +1,181 @@
+/**
+ * Copyright (c) 2014 openHAB UG (haftungsbeschraenkt) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.vitotronic.handler;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.smarthome.core.items.Item;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link VitotronicHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ * 
+ * @author Stefan Andres - Initial contribution
+ */
+
+public class VitotronicThingHandler extends BaseThingHandler implements VitotronicThingListner {
+
+    private Logger logger = LoggerFactory.getLogger(VitotronicThingHandler.class);
+    
+    private int pollingBridge = 60; // every seconds ask for Bridge
+    VitotronicBridgeHandler bridgeHandler;
+
+
+	public VitotronicThingHandler(Thing thing) {
+		super(thing);
+	}
+	
+	@Override
+	public void initialize() {
+		logger.debug("Thing Handler for {} started", getThing().getUID().getId() );
+		bridgeOnlineWatchdog();
+	}
+	
+	@Override
+	public void dispose() {
+		logger.debug("Thing Handler for {} stop", getThing().getUID().getId() );
+		unregisterThingListener();
+	}
+
+	@Override
+	public void handleCommand(ChannelUID channelUID, Command command) {
+		logger.trace("Handle command for channel '{}' command '{}'", channelUID.getId(), command.toString());
+		bridgeHandler.updateChannel(getThing().getUID().getId(), channelUID.getId(), command.toString());
+     
+	}
+	
+	private void bridgeOnlineWatchdog() {

```

Comment:
instead of this watchdog, couldn't the bridge handler directly update the thing status of the registered handlers? This would avoid having to poll for status updates.